api/src/main/java/io/druid/data/input/InputRow.java

```diff
@@ -28,8 +30,8 @@
  * implement "schema-less" data ingestion that allows the system to add new dimensions as they appear.
  *
  */
-public interface
-    InputRow extends Row
+@PublicApi
```

Comment:
hmmm, so I took a brief look at avro and orc parsers which create MapBasedInputRow and it looks like it would be better to create an impl of InputRow instead .

to use MapBasedInputRow..
- avro parser makes GenericRecord look like a Map here ( https://github.com/druid-io/druid/blob/master/extensions-core/avro-extensions/src/main/java/io/druid/data/input/avro/GenericRecordAsMap.java ) and most methods have "unimplemented exceptions" , would probably be better to just make GenericRecord look like an InputRow.

- orc parser converts its ord record into a map here ( https://github.com/druid-io/druid/blob/master/extensions-contrib/orc-extensions/src/main/java/io/druid/data/input/orc/OrcHadoopInputRowParser.java#L78 ) which would be inefficient if your input record had alot of columns that were not being indexed into druid. again it would be better to make orc record look like an InputRow instead.

with that information, what do you think ?